src/controllers/messages.ts

```diff
@@ -797,6 +797,72 @@ export const receiveDeleteMessage = async (payload: Payload): Promise<void> => {
   )
 }
 
+/**
+Either makes all messages of a chat read or it marks the last message of the given chat as unread.
+
+@param {Req} req - The request object containing the chat ID.
+@param {Res} res - The response object used to send the updated chat information.
+@returns {Promise<void>} - An empty promise.
+*/
+export const toggleChatReadUnread = async (req: Req, res: Res): Promise<void> => {
+  if (!req.owner) return failure(res, 'no owner')
+
+  try{
+    const requestBody = JSON.parse(req.rawBody);
+      //We want to mark the chat as read
+    if (requestBody.shouldMarkAsUnread === false) {
+      // Call the readMessages function with the same req and res objects
+      const result = await readMessages(req, res);
+      return result; // Exit the function after calling readMessages
+    }
+
+    //we want to mark the last message + chat as unread
+    const chat_id = req.params.chat_id
+    const owner = req.owner
+    const tenant: number = owner.id
+
+    // Find the latest message that is not from the user and has been read
+    const latestMessage = await models.Message.findOne({
+      where: {
+        sender: {
+          [Op.ne]: owner.id,
+        },
+        chatId: chat_id,
+        [Op.or]: [{ seen: true }, { seen: null }],
+        tenant,
+      },
+      order: [['date', 'DESC']],
+    });
+
+    // Update just the latest message found in the database to mark it as unseen
+    if (latestMessage) {
+      await latestMessage.update({ seen: false });
+    }
+    //mark the chat as unseen
+    const chat: Chat = (await models.Chat.findOne({
+      where: { id: chat_id, tenant },
+    })) as Chat
+    if (chat) {
+      resetNotifyTribeCount(parseInt(chat_id))
+      await chat.update({ seen: false })
+      success(res, {})
+      sendNotification(chat, '', 'badge', owner)
```

Comment:
why are we sending a push notification here? Weird to send a notification for something we do manually

Reply:
good point. I went ahead and removed that line.